02-express-tutorial/app.js

```diff
@@ -1,60 +1,137 @@
-//import express modules and data
-const express = require("express");
-const { products } = require("./data");
-const app = express();
-const port = 3000;
-
-//middleware for stattic files from 'public' directory
-app.use(express.static("./public"));
-
-//routes for /api/v1/test to return JSON
-app.get("/api/v1/test", (req, res) => {
-    res.json({ message: "It worked!" });
-  });
-
-//routes to return all products
-app.get("/api/v1/products", (req, res) => {
-    res.json(products);
-  });
-
-//get product by ID
-app.get("/api/v1/products/:productID", (req, res) => {
-    const idToFind = parseInt(req.params.productID);
-    const product = products.find((p) => p.id === idToFind);
-  
-    if (!product) {
-      return res.status(404).json({ message: "That product was not found." });
-    }
-  
-    res.json(product);
-  });
-
-// query products
-app.get("/api/v1/query", (req, res) => {
-    const { search, limit } = req.query;
-    let filteredProducts = products;
-  
-    if (search) {
-      filteredProducts = filteredProducts.filter((product) =>
-        product.name.toLowerCase().startsWith(search.toLowerCase())
-      );
-    }
-  
-    if (limit) {
-      filteredProducts = filteredProducts.slice(0, parseInt(limit));
+
+const express = require('express')
+const app = express()
+const logger = require('./logger')
+const { products, people } = require("./data");
+const peopleRouter = require("./routes/people");
+const cookieParser = require("cookie-parser");
+
+//apply logger for single route
+app.get('/', logger, (req, res) => {
+  res.send('Home')
+})
+app.get('/about', logger, (req, res) => {
+  res.send('About')
+})
+app.get('/api/products', logger, (req, res) => {
+  res.send('Products')
+})
+
+//apply logger globally
+app.use(logger)
+app.get('/',  (req, res) => {
+  res.send('Home')
+})
+app.get('/about', (req, res) => {
+  res.send('About')
+})
+app.get('/api/products', (req, res) => {
+  res.send('Products')
+})
+
+//GET /api/v1/people request:
+app.get('/api/v1/people', (req, res) => {
+  res.json(people);
+});
```

Comment:
Duplicated endpoint. This one, since it comes first, is called instead of the one defined in the people router.